mqtt/mqtt-broker/src/broker.rs

```diff
@@ -622,11 +629,14 @@ where
                 let (new_session, events, session_present) =
                     if let proto::ClientId::IdWithExistingSession(_) = connreq.connect().client_id {
                         debug!("moving offline session to online for {}", client_id);
-                        let (state, events) = offline
-                            .into_online()
-                            .map_err(|_| SessionError::PacketIdentifiersExhausted)?;
-                        let new_session = Session::new_persistent(auth_id, connreq, state);
-                        (new_session, events, true)
+                        if let Ok((state, events)) = offline.into_online() {
+                            let new_session = Session::new_persistent(auth_id, connreq, state);
+                            (new_session, events, true)
+                        } else {
+                            panic!(
```

Comment:
there was no panic before. can we possibly revert this change back?

Reply:
there was, it just moved